### DISCUSSION TITLE
noUselessElse is bad in some cases

### DISCUSSION THREAD

**Discussion by Jack-Works** [biomejs/biome#2581] [General]
For code

```js
function f() {
    if (x) {
        // lines of code
    } else if (y) {
        // lines of code
    } else if (z) {
        // lines of code
    } else {
        // lines of code
    }
}

function f2() {
    if (x) {
        // lines of code
    }
    if (y) {
        // lines of code
    }
    if (z) {
        // lines of code
    }
    // lines of code
}
```

This rule harms the readability in this case.

You can know that `f` only falls into one `if` branch because all branches are `if, else if, else` with only a glimpse, but you cannot get that information from `f2`. You need to analyze `f2` to understand what it does.

**Comment by ematipico** [comment]
We picked this up from another ecosystem and were happy with the results. Readability is a subjective matter, and there is no right answer. You can still disable the rule if you don't want it.
